app/models/course.rb

```diff
@@ -283,7 +285,11 @@ def tracked_revisions
   end
 
   def tracked_namespaces
-    return [0]
+    course_wiki_namespaces.map do |course_wiki_ns|
+      wiki = course_wiki_ns.courses_wikis.wiki
+      namespace = course_wiki_ns.namespace
+      { :wiki => wiki, :namespace => namespace }
```

Comment:
I think it will be important for the fallback/default behavior of tracking mainspace to be defined here, rather than (for example)  in ArticlesCourses#get_revisions_by_namespaces. Otherwise, we may end up needing to account for it in multiple places independently.

As you pointed out, I didn't account for this needing to return an array wiki+namespace combinations rather than just an array of namespace integers.

Here's how I think it should work: for each tracked wiki, if there is a course_wiki_namespace record, add a hash for that Wiki+Namespace, and otherwise, include a hash for Wiki+Mainspace.

Reply:
@ragesoss, I have updated it accordingly.